engine_graphics/src/Renderer.cpp

```diff
@@ -0,0 +1,463 @@
+    glClear(GL_COLOR_BUFFER_BIT);
+
+    const auto &vertices = mesh.vertices();
+
+    glColor3f(1.0, 1.0, 1.0);
+    //TODO: compute ortho
+    glOrtho(-8.0, 8.0, -4.5, 4.5, -1.0, 1.0);
+
+    for (auto cit = mesh.facets_cbegin(); cit != mesh.facets_cend(); ++cit)
+    {
+        const auto &tri = *cit;
+
+        const auto &p1 = vertices.at(tri[0]);
+        const auto &p2 = vertices.at(tri[1]);
+        const auto &p3 = vertices.at(tri[2]);
+
+        std::cout << p1 << std::endl;
+        std::cout << p2 << std::endl;
+        std::cout << p3 << std::endl;
+
+        // TODO: implement scaler
+        glBegin(GL_TRIANGLES);
+            glVertex3f(p1.x() / 100, p1.y() / 100, p1.z() / 100);
+            glVertex3f(p2.x() / 100, p2.y() / 100, p2.z() / 100);
+            glVertex3f(p3.x() / 100, p3.y() / 100, p3.z() / 100);
+        glEnd();
+    }
+
+    glFlush();
+}
+
+RendererImplementation::RendererImplementation(RendererConfig &&config)
+:
+_config(config)
+{
+    RendererImplementation::_vertex_buffers.push_back(0);
+    RendererImplementation::_inbex_buffer.push_back(0);
+}
+
+RendererImplementation::~RendererImplementation()
+{
+    _rendering_buffer.clear();
+    for (auto &v : RendererImplementation::_vertex_buffers)
+    {
+        glDeleteBuffers(1, &v);
+    }
+
+    for (auto &i : RendererImplementation::_inbex_buffer)
+    {
+        glDeleteBuffers(1, &i);
+    }
+
+    RendererImplementation::_inbex_buffer.clear();
+    RendererImplementation::_vertex_buffers.clear();
+    RendererImplementation::_vertices.clear();
+    RendererImplementation::_indices.clear();
+}
+
+
+void RendererImplementation::initGL()
+{
+    glShadeModel(GL_SMOOTH);                    // shading mathod: GL_SMOOTH or GL_FLAT
+    glPixelStorei(GL_UNPACK_ALIGNMENT, 4);      // 4-byte pixel alignment
+
+    // enable /disable features
+    glHint(GL_PERSPECTIVE_CORRECTION_HINT, GL_NICEST);
+    //glHint(GL_LINE_SMOOTH_HINT, GL_NICEST);
+    //glHint(GL_POLYGON_SMOOTH_HINT, GL_NICEST);
+    glEnable(GL_DEPTH_TEST);
+    glEnable(GL_LIGHTING);
+    glEnable(GL_TEXTURE_2D);
+    glEnable(GL_CULL_FACE);
+
+     // track material ambient and diffuse from surface color, call it before glEnable(GL_COLOR_MATERIAL)
+    glColorMaterial(GL_FRONT_AND_BACK, GL_AMBIENT_AND_DIFFUSE);
+    glEnable(GL_COLOR_MATERIAL);
+
+    glClearColor(_config.clear_color.r(), _config.clear_color.g(), _config.clear_color.b(), _config.clear_color.a());                   // background color
+    glClearStencil(0);                          // clear stencil buffer
+    glClearDepth(1.0f);                         // 0 is near, 1 is far
+    glDepthFunc(GL_LEQUAL);
+
+    initLights();
+}
+
+void RendererImplementation::initGLUT()
+{
+    // GLUT stuff for windowing
+    // initialization openGL window.
+    // it is called before any other GLUT routine
+    int i = 1;
+    char * argv[1];
+    argv[0] = const_cast<char*>( _config.app_path.c_str());
```

Comment:
Test if this can be removed, and if it can, remove it.

Reply:
Changed to caption instead of path, but shitty c code had to be left.